src/server/scripts/Pet/pet_hunter.cpp

```diff
@@ -95,20 +95,20 @@ class npc_pet_hunter_snake_trap : public CreatureScript
                     Unit* summoner = me->ToTempSummon()->GetSummonerUnit();
 
                     std::vector<Unit*> targets;
-                    for (std::pair<ObjectGuid const, PvPCombatReference*> const& pair : summoner->GetCombatManager().GetPvPCombatRefs())
+
+                    auto check = [&](auto pair) mutable
```

Comment:
```suggestion
                    auto addTargetIfValid = [&](CombatReference const* ref)
```

Reply:
The parameter can't be `CombatReference const* ref` if I will not edit the loops, because there is 
```cpp 
std::pair<ObjectGuid const, PvPCombatReference*> const& pair
``` 
```cpp
std::pair<ObjectGuid const, CombatReference*> const& pair
```